src/main/kotlin/org/wa/usfmreader/persistence/dao/BookDaoImpl.kt

```
@@ -0,0 +1,100 @@
+package org.wa.usfmreader.persistence.dao
+
+import io.reactivex.Completable
+import io.reactivex.Maybe
+import io.reactivex.Single
+import jooq.Tables
+import jooq.tables.daos.BookDao
+import org.wa.usfmreader.data.entities.BookData
+import org.wa.usfmreader.data.entities.LanguageData
+import org.wa.usfmreader.persistence.db.SqliteDB
+import org.wa.usfmreader.persistence.mappers.BooksDaoMapper
+
+class BookDaoImpl: AppBookDao {
+    private val db = SqliteDB()
+    private val bookDao = BookDao(db.configuration())
+    private val mapper = BooksDaoMapper()
+
+    override fun fetchOneBySlug(slug: String): Maybe<BookData> {
+        return Maybe.fromCallable {
+            mapper.mapFromBook(
+                    bookDao.fetchBySlug(slug).first()
+            )
+        }
```

Could this maybe throw an error if no book exists? If so, might be a good idea to go ahead and `.onErrorComplete()` here

---

Well I'm not using this method in the app. But ther is another one that I'm using "fetchOneByBookAndLanguage", but I can't put onErrorComplete, because I'm using onErrorResumeNext, to get the book from api, if it doesn't exist locally

---

Okay, that sounds good. If you aren't using the method, maybe just remove it then